mobile/src/main/java/org/openhab/habdroid/ui/OpenHABMainActivity.java

```diff
@@ -242,6 +247,24 @@ protected void onCreate(Bundle savedInstanceState) {
         // Set the theme to one from preferences
         mSettings = PreferenceManager.getDefaultSharedPreferences(this);
 
+        //  Create a new boolean and preference and set it to true
+        boolean isFirstStart = mSettings.getBoolean("firstStart", true);
+
+        //  If the activity has never started before...
+        if (isFirstStart) {
+
+            //  Launch app intro
+            final Intent i = new Intent(OpenHABMainActivity.this, IntroActivity.class);
+
+            runOnUiThread(new Runnable() {
+                @Override public void run() {
+                    startActivity(i);
+                }
+            });
+
```

Comment:
onCreate runs on the UI thread, no need for runOnUiThread here

Reply:
Ahh, I just copied from here https://github.com/apl-devs/AppIntro/wiki/How-to-Use#show-the-intro-once

Reply:
Ok, I just see he creates a new thread.